jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java

```diff
@@ -68,7 +68,8 @@ public abstract class SecurityHandler extends HandlerWrapper implements Authenti
     private final Map<String, String> _initParameters = new HashMap<>();
     private LoginService _loginService;
     private IdentityService _identityService;
-    private boolean _renewSession = true;
+    private boolean _renewSessionOnAuthentication = true;
+    private int _sessionMaxInactiveIntervalOnAuthentication = 0;
```

Comment:
I don't think a default of 0 is a good idea. If it is not set, then sessions that were previously configured - either programmatically or via web.xml -  to timeout after x minutes are suddenly immortal after authentication.

Ooops, commented too soon. Now I've read the `LoginAuthenticator` code and it's the other way around: you can't cause sessions to become immortal after being authenticated. Immortality is  indicated by a timeout <= 0,  but you're only setting the maxInactiveInterval iff it's >  0.